master/internal/api_auth.go

```diff
@@ -139,8 +139,9 @@ func processProxyAuthentication(c echo.Context) (done bool, err error) {
 
 	var ok bool
 	if spec.TaskType == model.TaskTypeTensorboard {
+		// TODO (eliu): validate access for all experiment workspaces that this tensorboard uses
 		ok, err = command.AuthZProvider.Get().CanGetTensorboard(
-			ctx, *user, spec.WorkspaceID)
+			ctx, *user, spec.WorkspaceID, nil, nil)
```

Comment:
this seems simplified now that the logic is shifted into CanGetTensorboard. this could be a matter of including exp and trial list to this call by updating what `db.IdentifyTask(ctx, taskID)` returns. also good with leaving out for a separate pr

Reply:
yeah was planning to leave this for a separate PR and push this one out quicker